File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder();
        int n = s.length();
        if(n == 1 || n == 2)
            return s;
        sb.append(s.charAt(0));
        sb.append(s.charAt(1));

        for(int i = 2; i < n; i += 1){
            int sz = sb.length();
            if(s.charAt(i) != sb.charAt(sz-1) || s.charAt(i) != sb.charAt(sz-2 )){
                sb.append(s.charAt(i));
            }
        }
        return sb.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res;

        for(int i =0; i< s.size(); i++){
            if(i-1 >=0 && i-2 >= 0 && res[res.size()-2]==s[i] && res[res.size()-1]==s[i]){
                continue;
            }
            res.push_back(s[i]);
        }

        return res;
    }
};